### PR TITLE
Apply url encoding to single header example, matching multiple

### PR DESCRIPTION
**Pull request by anuraaga**
***
<a href="https://pr-preview.s3.amazonaws.com/anuraaga/correlation-context/pull/78.html" title="Last updated on Dec 8, 2021, 4:07 PM UTC (2060adb)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/baggage/78/f45f946...anuraaga:2060adb.html" title="Last updated on Dec 8, 2021, 4:07 PM UTC (2060adb)">Diff</a>

### CODE REVIEW

File: baggage/HTTP_HEADER_FORMAT.md [hunk 1]
```diff
@@ -86,32 +86,34 @@ baggage: key1=value1;property1;property2, key2 = value2, key3=value3; propertyKe
 
 ## Examples of HTTP headers
 
+When propagating the entries (userId, alice), (serverNode, DF 28), (isProduction, false),
```
[the remark this answers is not in the document]
Comment (yurishkuro) [reply baggage/HTTP_HEADER_FORMAT.md:89]: comma-tuples are confusing to read, I suggest
```suggestion
Assume we want to propagate these entries: `userId="alice"`, `serverNode="DF 28"`, `isProduction=false`,
```

[the remark this answers is not in the document]
Comment (yurishkuro) [reply baggage/HTTP_HEADER_FORMAT.md:89]: > there are two URL encodings for it

what do you mean?

[the remark this answers is not in the document]
Comment (anuraaga) [reply baggage/HTTP_HEADER_FORMAT.md:89]: > comma-tuples are confusing to read, I suggest

Sure just didn't want to get it mixed up with the actual encoding which looks similar.

[the remark this answers is not in the document]
Comment (anuraaga) [reply baggage/HTTP_HEADER_FORMAT.md]: > what do you mean?

In query-parameter encoding, space is encoded as `+`, while in url path encoding, space is encoded as `%20`. My understanding is, lacking any other specificity, `%20` is the correct one to use.

For reference, https://github.com/open-telemetry/opentelemetry-java/pull/3711#discussion_r724628148

**Review by yurishkuro — APPROVED** [review]

File: baggage/HTTP_HEADER_FORMAT.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (yurishkuro) [reply baggage/HTTP_HEADER_FORMAT.md]: > L53: A value contains a URL encoded UTF-8 string.

should we change this text then to be more specific which URL encoding is meant?

[the remark this answers is not in the document]
Comment (yurishkuro) [reply baggage/HTTP_HEADER_FORMAT.md]: There's probably an RFC for it

**Comment by plehegar** [comment]
(attempting to redeliver the GH payload)

**Pull request merged**